CONTRIBUTING.md

```diff
@@ -0,0 +1,140 @@
+# Contributing
+
+Thank you for your interest in contributing to tendermint-rs. The goal
+of tendermint-rs is to provide a high quality, formally verified implementation of
+Tendermint.
+
+All work on the code base should be motivated by a Github
+Issue. Search is a good place start when looking for places to contribute. If you
+would like to work on an issue which already exists, please indicate so
+by leaving a comment. If you'd like to work on something else, open an Issue to
+start the discussion.
+
+The rest of this document outlines the best practices for contributing to this
+repository:
+
+- [Decision Making](#decision-making) - process for agreeing to changes
+- [Forking](#forking) - fork the repo to make pull requests
+- [Changelog](#changelog) - changes must be recorded in the changelog
+- [Pull Requests](#pull-requests) - what makes a good pull request
+
+## Decision Making
+
+When contributing to the project, the following process leads to the best chance of
+landing to changes in master.
+
+All new contributions should start with a Github
+Issue. The issue helps capture the problem you're trying to solve and allows for
+early feedback. Once the issue is created, maintainers may request more detailed
+documentation be written in the form of a Request for Comment (RFC) or
+Architectural Decision Record (ADR).
```

Comment:
Maybe link to https://github.com/informalsystems/tendermint-rs/blob/master/docs/architecture/README.md (?)
Not sure if there is an RFC equivalent.